### ISSUE TITLE
Reproducing the Results of Table 2

### ISSUE THREAD

**Issue by Tree-Shu-Zhao** [IDEA-FinAI/RagLLaVA#18]
Thanks for your work. I am wondering how to reproduce the results of WebQA in Table 2, first row. I adopted the pre-trained CLIP-ViT-L/14-336px and used Faiss to build an index for all images.

Then, I used around 5000 questions from the dev set as queries and retrieved the top 2 (and top 5) images. Using the evaluation code from https://github.com/IDEA-FinAI/RagVL/blob/main/webqa_pipeline.py, the precision, recall, and F1 can only achieve 5.2, 21.4, and 8.3.

**Comment by kzhpchen** [comment]
You need to make sure the `val_dataset`, `index`, and `index_to_image_id` are matched. The results of WebQA in Table 2 are from the test set we provided, btw.

**Comment by Tree-Shu-Zhao** [comment]
@SakuraTroyChen Thanks. I double-checked the code and data. I found the WebQA_test_image_index_to_id.json and WebQA_dev_image_index_to_id.json contains 3353 and 4972 images. I use the following code to obtain the number of images.

```python
import json

with open("WebQA_test.json", "r") as f:
    data = json.load(f)

test_image_ids = set()
for guid in data:
    item = data[guid]
    img_infos = item["img_Facts"]
    for img_info in img_infos:
        image_id = img_info["image_id"]
        test_image_ids.add(image_id)
print(f"#Test Images: {len(test_image_ids)}")

train_image_ids = set()
val_image_ids = set()
with open("WebQA_train_val.json", "r") as f:
    data = json.load(f)
for guid in data:
    item = data[guid]
    split = item["split"]
    img_infos = item["img_posFacts"] + item["img_negFacts"]
    for img_info in img_infos:
        image_id = img_info["image_id"]
        if split == "train":
            train_image_ids.add(image_id)
        elif split == "val":
            val_image_ids.add(image_id)
print(f"#Train Images: {len(train_image_ids)}")
print(f"#Val Images: {len(val_image_ids)}")
print(f"#Total Images: {len(train_image_ids.union(val_image_ids).union(test_image_ids))}")
```

The number of train/val/test images should be 322345, 64125, and 90760, respectively. The total number of images in the retrieval dataset should be 389750, which is aligned with the number reported in the WebQA paper.

I am wondering why there are thousands of images in image_index_to_id.json files and how should I select them from the original dataset?

**Comment by kzhpchen** [comment]
@Tree-Shu-Zhao As declared in Section 4.1 of our paper, we only consider the image-related subsets of WebQA and MultimodalQA. Since the test set labels from both datasets are not publicly available, we report the results on the validation set, which is the test set in our setting. Thus, to reproduce the results, you should use the test set from [Google Drive](https://drive.google.com/drive/folders/1wY18Vbrb8yDbFSg1Te-FQIs84AYYh48Z?usp=drive_link).

Hope your problem has been resolved. Thanks!
